storage/remote/queue_manager.go

```diff
@@ -552,13 +555,19 @@ func (s *shards) start(n int) {
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
 
-	newQueues := make([]chan prompb.TimeSeries, n)
+	newQueues := make([]chan sample, n)
 	for i := 0; i < n; i++ {
-		newQueues[i] = make(chan prompb.TimeSeries, s.qm.cfg.Capacity)
+		newQueues[i] = make(chan sample, s.qm.cfg.Capacity)
 	}
 
 	s.queues = newQueues
 
+	s.tsPool = sync.Pool{
```

Comment:
Does this really help? Changing the number of shards isn't that often.

Reply:
Benchmarks improved some when I first added it. The pool is used for every http request, not just the number of shards, though it could probably be done by keeping a single `[]prompb.TimeSeries` per shard instead of using a pool, similar to the compression buffer.